addons/project/controllers/portal.py

```diff
@@ -224,6 +224,18 @@ def portal_my_project_task(self, project_id=None, task_id=None, access_token=Non
         task_sudo.attachment_ids.generate_access_token()
         values = self._task_get_page_view_values(task_sudo, access_token, project=project_sudo, **kw)
         values['project'] = project_sudo
+
+        history = request.session.get('my_project_tasks_history', [])
+        try:
+            current_task_index = history.index(task_id)
+        except ValueError:
+            return request.render("project.portal_my_task", values)
+
+        total_task = len(history)
+        task_url = f"/my/project/{project_id}/task/%s?model=project.project&res_id={values['user'].id}&access_token={access_token}"
+        values['prev_record'] = task_url % history[(current_task_index - 1) % total_task] if total_task > 1 else False
+        values['next_record'] = task_url % history[(current_task_index + 1) % total_task] if total_task > 1 else False
```

Comment:
Just so you know, you can use short-circuits instead of ternary operators where one of the results would be a boolean, like this:
```suggestion
        values['prev_record'] = total_task > 1 and task_url % history[(current_task_index - 1) % total_task]
        values['next_record'] = total_task > 1 and task_url % history[(current_task_index + 1) % total_task]
```
It's not always the best readability-wise though, so I leave it up to you

Reply:
Indeed when mixing return types I personally find ternary operators more readable.

Reply:
i think so too, but almost all of the code of odoo use shortcuts, so i thought it would be better to be in line with the rest of the code.